pkg/postage/postagecontract/contract.go

```diff
@@ -301,6 +302,70 @@ func (c *postageContract) getBalance(ctx context.Context) (*big.Int, error) {
 	return abi.ConvertType(results[0], new(big.Int)).(*big.Int), nil
 }
 
+func (c *postageContract) getLastPrice(ctx context.Context) (uint64, error) {
+	callData, err := c.postageStampContractABI.Pack("lastPrice")
+	if err != nil {
+		return 0, err
+	}
+
+	result, err := c.transactionService.Call(ctx, &transaction.TxRequest{
+		To:   &c.postageStampContractAddress,
+		Data: callData,
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	results, err := c.postageStampContractABI.Unpack("lastPrice", result)
+	if err != nil {
+		return 0, err
+	}
+
+	if len(results) == 0 {
+		return 0, errors.New("unexpected empty results")
+	}
+
+	return results[0].(uint64), nil
+}
+
+func (c *postageContract) getMinimumValidityBlocks(ctx context.Context) (uint64, error) {
+	callData, err := c.postageStampContractABI.Pack("minimumValidityBlocks")
+	if err != nil {
+		return 0, err
+	}
+
+	result, err := c.transactionService.Call(ctx, &transaction.TxRequest{
+		To:   &c.postageStampContractAddress,
+		Data: callData,
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	results, err := c.postageStampContractABI.Unpack("lastPrice", result)
```

Comment:
this is unpacking results from a `lastPrice` call even though `minimumValidityBlocks` was packed above..
why haven't the unit tests caught up on this bug?
let's please improve the unit tests

Reply:
Coz the transaction mock does not care about the unpak name. So it didn't catch it. I don't see any other way to improve the test.